plugins/hydra_orion_sweeper/hydra_plugins/hydra_orion_sweeper/_impl.py

```diff
@@ -0,0 +1,347 @@
+        trials = []
+
+        for _ in range(count):
+            try:
+                trial = self.client.suggest(pool_size=count)
+                trials.append(trial)
+
+            # non critical errors
+            except WaitingForTrials:
+                break
+
+            except ReservationRaceCondition:
+                break
+
+            except CompletedExperiment:
+                break
+
+        return trials
+
+    def new_experiment(self, arguments) -> ExperimentClient:
+        """Initialize orion client from the config and the arguments"""
+
+        self.space_parser.add_from_overrides(arguments)
+        self.space, self.arguments = self.space_parser.space()
+
+        dict_config = OmegaConf.to_container(self.algo_config)
+        algo_type = dict_config.pop("type", "random")
+        algo_config = dict_config.pop("config", dict())
+
+        log.info("Orion Optimizer %s", self.algo_config)
+        log.info("with parametrization %s", self.space.configuration)
+
+        return create_experiment(
+            name=self.orion_config.name,
+            version=self.orion_config.version,
+            space=self.space,
+            algorithms={algo_type: algo_config},
+            strategy=None,
+            max_trials=self.worker_config.max_trials,
+            max_broken=self.worker_config.max_broken,
+            storage=self.storage_config,
+            branching=self.orion_config.branching,
+            max_idle_time=None,
+            heartbeat=None,
+            working_dir=None,
+            debug=self.orion_config.debug,
+            executor=None,
+        )
+
+    def sweep(self, arguments: List[str]) -> None:
+        """Execute the optimization process"""
+
+        assert self.config is not None
+        assert self.launcher is not None
+        assert self.job_idx is not None
+
+        log.debug("Starting new experiment")
+        self.client = self.new_experiment(arguments)
+
+        with clientctx(self.client):
+            try:
+                self.optimize(self.client)
+            except Exception as e:
+                self.release_all()
+                raise e
+
+    def release_all(self) -> None:
+        """Make sure not trials remain reserved"""
+        for trial in self.pending_trials:
+            try:
+                self.client.release(trial, status="interrupted")
+            except AlreadyReleased:
+                pass
+
+    def optimize(self, client: ExperimentClient) -> None:
+        """Run the hyperparameter search in batches"""
+        failures = []
+
+        while not self.client.is_done:
+            trials = self.sample_trials()
+
+            if len(trials) == 0 and not self.client.is_done and not self.pending_trials:
+                raise RuntimeError()
+
+            returns = self.execute_trials(trials)
+
+            self.observe_results(trials, returns, failures)
+
+            if len(failures) > self.worker_config.max_broken:
```

Comment:
```suggestion
            if self.client.is_broken:
```
This would better handle resumption (if possible with Hydra).

Reply:
But we may have no failures if user is trying to resume a broken experiment...

Reply:
if the client is broken that means some workers had failures, that worker will raise the exception.
In that case if failures == 0 should we just log an error and not raise ?

Reply:
to avoid printing to many exception with no debug info

Reply:
I think it would make sense to raise BrokenExperiment anyhow. But the stack-trace will not show one of the exceptions from previous worker execution.